mapswipe_workers/mapswipe_workers/firebase_to_postgres/update_data.py

```diff
@@ -88,42 +91,98 @@ def update_user_data(user_ids: list = []) -> None:
     logger.info("Updated user data in Potgres.")
 
 
-def update_project_data(project_ids: list = []):
+def get_project_attribute_from_firebase(project_ids: List[str], attribute: str):
+    """Use async for firebase sdk to query a project attribute.
+
+    Follows a workflow describted in this blogpost:
+    https://hiranya911.medium.com/firebase-python-admin-sdk-with-asyncio-d65f39463916
     """
-    Gets status of projects
-    from Firebase and updates them in Postgres.
 
-    Default behavior is to update all projects.
+    async def get_project_status(project_id, event_loop):
+        ref = fb_db.reference(f"v2/projects/{project_id}/{attribute}")
+        # Blocking method is delegated to the thread pool
+        return [project_id, await event_loop.run_in_executor(executor, ref.get)]
+
+    async def get_project_status_list(project_ids, event_loop):
+        coroutines = [get_project_status(i, event_loop) for i in project_ids]
+        completed, pending = await asyncio.wait(coroutines)
+        firebase_status_dict = {}
+        for item in completed:
+            project_id, firebase_status = item.result()
+            firebase_status_dict[project_id] = firebase_status
+        return firebase_status_dict
+
+    fb_db = auth.firebaseDB()
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=20)
```

Comment:
I'm not sure if I understand this question properly.

Reply:
I think what @laurentS whats to point out is that `executer` is not passed as argument to `get_project_status` but it is still used inside this function. This will work though. `executer` is in scope for `get_project_status`.